tools/cmake/mbed_greentea.cmake

```diff
@@ -55,11 +55,7 @@ macro(mbed_greentea_add_test)
             ${MBED_GREENTEA_TEST_SOURCES}
     )
 
-    if(MBED_BAREMETAL_GREENTEA_TEST)
-        list(APPEND MBED_GREENTEA_TEST_REQUIRED_LIBS mbed-baremetal mbed-greentea)
-    else()
-        list(APPEND MBED_GREENTEA_TEST_REQUIRED_LIBS mbed-os mbed-greentea)
-    endif()
+    list(APPEND MBED_GREENTEA_TEST_REQUIRED_LIBS mbed-greentea)
```

Comment:
We should unify, `MBED_TEST_LINK_LIBRARIES` should be the list to be used. This greentea cmake would add mbed-greentea  to it. An app would add `mbed-os/mbed-baremetal` and other libs.